db/crud.go

```diff
@@ -896,34 +896,42 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 }
 
 // updateHLV updates the HLV in the sync data appropriately based on what type of document update event we are encountering
-func (db *DatabaseCollectionWithUser) updateHLV(d *Document, docUpdateEvent DocUpdateType) (*Document, error) {
+func (db *DatabaseCollectionWithUser) updateHLV(d *Document, previousMou *MetadataOnlyUpdate, docUpdateEvent DocUpdateType) (*Document, error) {
 
+	hasHLV := d.HLV != nil
 	if d.HLV == nil {
 		d.HLV = &HybridLogicalVector{}
 	}
 	switch docUpdateEvent {
 	case ExistingVersion:
 		// preserve any other logic on the HLV that has been done by the client, only update to cvCAS will be needed
 		d.HLV.CurrentVersionCAS = expandMacroCASValueUint64
-		d.HLV.ImportCAS = 0 // remove importCAS for non-imports to save space
 	case Import:
-		if d.HLV.CurrentVersionCAS == d.Cas {
-			// if cvCAS = document CAS, the HLV has already been updated for this mutation by another HLV-aware peer.
-			// Set ImportCAS to the previous document CAS, but don't otherwise modify HLV
-			d.HLV.ImportCAS = d.Cas
-		} else {
+		// Update HLV if:
```

Comment:
I think the description as you've got it muddies the rationale behind the decision making here (simply that we want to update the HLV only if the current document cas isn't already accounted for in the HLV + mou).  
```suggestion
		// Do not update HLV if the current document version (cas) is already included in the existing HLV, 
		// as either:
		//    1. _vv.cvCAS == document.cas (current mutation is already present as cv), or
		//    2. _mou.cas == document.cas (current mutation is already present as cv, and was imported on a different cluster)
```